ee/clickhouse/views/events.py

```diff
@@ -1,34 +1,50 @@
 import json
+from typing import Any, Dict, List, Optional
 
 from rest_framework import viewsets
 from rest_framework.decorators import action
 from rest_framework.request import Request
 from rest_framework.response import Response
 
 from ee.clickhouse.client import sync_execute
-from ee.clickhouse.models.element import get_elements_by_elements_hash
+from ee.clickhouse.models.element import get_elements_by_elements_hash, get_elements_by_elements_hashes
 from ee.clickhouse.models.event import ClickhouseEventSerializer, determine_event_conditions
 from ee.clickhouse.models.property import get_property_values_for_key, parse_filter
+from ee.clickhouse.queries.util import parse_timestamps
 from ee.clickhouse.sql.events import SELECT_EVENT_WITH_ARRAY_PROPS_SQL, SELECT_EVENT_WITH_PROP_SQL, SELECT_ONE_EVENT_SQL
 from ee.clickhouse.util import CH_EVENT_ENDPOINT, endpoint_enabled
 from posthog.api.event import EventViewSet
-from posthog.models.filter import Filter
+from posthog.models import Filter, Team
 from posthog.utils import convert_property_value
 
 
 class ClickhouseEvents(EventViewSet):
-    def list(self, request):
+    def _get_elements(self, query_result: List[Dict], team: Team) -> Dict[str, Any]:
+        element_hashes = [event[6] for event in query_result]
```

Comment:
I really don't like that we are pulling out hashes by index `6` here. Has a bit of a smell. Not a huge deal though. If we did this in a few spots it would make it really hard to refactor the query and reorder columns.

Would be better if we had `event.element_hash` on the row. Something we can work towards.

Reply:
Yeah agree, need to solve this. surprised the driver doesn't give you a dict or object